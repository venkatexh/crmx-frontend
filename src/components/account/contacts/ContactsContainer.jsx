import { useEffect } from "react";
import { useDispatch, useSelector } from "react-redux";
import { saveContacts } from "../../../redux/actions/account/saveContacts";
import "../../../sass/components/account/contacts/contactsContainer.component.scss";

const ContactsContainer = () => {
  const dispatch = useDispatch();
  const state = useSelector(({ loggedUser, contacts }) => ({
    loggedUser,
    contacts,
  }));
  useEffect(() => {
    setTimeout(() => {
      dispatch(saveContacts(state.loggedUser.id));
    }, 1000);
  }, []);
  return (
    <div
      className={`${
        state.contacts.length === 0 ? "loading" : "loaded"
      } container`}
    >
      {state.contacts.length === 0 ? (
        <div>
          <img
            src={"/loaders/comp_loader.gif"}
            alt={"loader"}
            style={{ height: "60px" }}
          />
        </div>
      ) : (
        <>
          <div className={"header"}>
            <div className={"colEmail"}>Email</div>
            <div className={"col"}>First Name</div>
            <div className={"col"}>Last Name</div>
            <div className={"col"}>Status</div>
            <div className={"col"}>Location</div>
          </div>
          <div className={"scroll"}>
            {state.contacts
              .slice(0)
              .reverse()
              .map(({ firstName, lastName, email, status, location }) => {
                return (
                  <div className={"rowContainer"}>
                    <div className={"contactRow"}>
                      <div className={"colEmail"}>{email}</div>
                      <div className={"col"}>{firstName}</div>
                      <div className={"col"}>{lastName}</div>
                      <div className={"col"}>{status}</div>
                      <div className={"col"}>{location}</div>
                    </div>
                    <hr className={"separator"} />
                  </div>
                );
              })}
          </div>
        </>
      )}
    </div>
  );
};

export default ContactsContainer;
