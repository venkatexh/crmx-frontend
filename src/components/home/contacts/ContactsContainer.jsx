import {useEffect, useState} from "react";
import {useDispatch, useSelector} from "react-redux";
import {saveContacts} from "../../../redux/actions/account/saveContacts";
import "../../../sass/components/home/contacts/contactsContainer.component.scss";
import Link from "react-router-dom/es/Link";
import {saveTags} from "../../../redux/actions/account/saveTags";

const ContactsContainer = () => {
  const [loading, setLoading] = useState(false);
  const dispatch = useDispatch();

  const state = useSelector(({loggedUser, contacts}) => ({
    loggedUser,
    contacts,
  }));

  useEffect(() => {
    setLoading(true);
    setTimeout(() => {
      dispatch(saveContacts(state.loggedUser.id));
      dispatch(saveTags(state.loggedUser.id))
      setLoading(false);
    }, 1000);
  }, []);

  return (
    <div className={`${loading ? "loading" : "loaded"} container`}>
      {loading ? (
        <div>
          <img
            src={"/loaders/comp_loader.gif"}
            alt={"loader"}
            style={{height: "60px"}}
          />
        </div>
      ) : state.contacts.length === 0 ? (
        <div>You don't have any contacts</div>
      ) : (
        <>
          <div className={"header"}>
            <div className={"colEmail"}>Email</div>
            <div className={"col"}>First Name</div>
            <div className={"col"}>Last Name</div>
            <div className={"col"}>Status</div>
            <div className={"col"}>Location</div>
          </div>
          <div className={"scroll"}>
            {state.contacts
              .slice(0)
              .reverse()
              .map(({_id, firstName, lastName, email, status, location}) => {
                return (
                  <Link to={`/contact/${_id}`} className={'contactLink'}>
                    <div key={_id} className={"rowContainer"}>
                      <div className={"contactRow"}>
                        <div className={"colEmail"}>{email}</div>
                        <div className={"col"}>{firstName}</div>
                        <div className={"col"}>{lastName}</div>
                        <div className={"col"}>{status}</div>
                        <div className={"col"}>{location}</div>
                      </div>
                      <hr className={"separator"}/>
                    </div>
                  </Link>
                );
              })}
          </div>
        </>
      )}
    </div>
  );
};

export default ContactsContainer;
