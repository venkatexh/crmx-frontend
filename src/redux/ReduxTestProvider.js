import { Provider } from "react-redux";

const ReduxTestProvider = ({ children, reduxStore }) => (
  <Provider store={reduxStore}>{children}</Provider>
);

export default ReduxTestProvider;
