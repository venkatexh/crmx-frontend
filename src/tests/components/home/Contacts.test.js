import { shallow } from "enzyme";
import React from "react";
import ContactsContainer from "../../../components/home/contacts/ContactsContainer";
import ContactsHeader from "../../../components/home/contacts/ContactsHeader";
import ReduxTestProvider from "../../../redux/ReduxTestProvider";

it("should be contacts container component snapshot", () => {
  expect(
    shallow(
      <ReduxTestProvider>
        <ContactsContainer />
      </ReduxTestProvider>
    )
  ).toMatchSnapshot();
});

it("should be contacts header component snapshot", () => {
  expect(
    shallow(
      <ReduxTestProvider>
        <ContactsHeader />
      </ReduxTestProvider>
    )
  ).toMatchSnapshot();
});
