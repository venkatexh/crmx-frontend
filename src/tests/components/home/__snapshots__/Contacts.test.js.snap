// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should be contacts container component snapshot 1`] = `ShallowWrapper {}`;

exports[`should be contacts header component snapshot 1`] = `ShallowWrapper {}`;
